### PR TITLE
slight refactor of rssi calc & remove a lot of f64 and unsigned usage

### PR DESCRIPTION
**Pull request by andymck**

### CODE REVIEW

File: iot_verifier/src/poc.rs [hunk 1]
```diff
@@ -590,11 +598,19 @@ fn verify_witness_data(beacon_data: &Vec<u8>, witness_data: &Vec<u8>) -> Generic
...
-fn calc_fspl(tx_power: i32, freq: u64, distance: f64, gain: i32) -> f64 {
-    let gt = 0.0;
-    let gl = gain as f64 / 10.0;
-    let fpsl = (20.0 * (4.0 * PI * distance * (freq as f64) / C).log10()) - gt - gl;
-    (tx_power as f64) - fpsl
+fn calc_expected_rssi(
+    conducted_tx_power_dbm: i32,
+    freq: u64,
+    distance_mtrs: u32,
+    beaconer_gain_dbm: i32,
```
Comment (ke6jjj) [inline iot_verifier/src/poc.rs:605]: Can you rename these parameters to:

`beaconer_gain_ddb` 
`witness_gain_ddb`

File: iot_verifier/src/poc.rs [hunk 2]
```diff
@@ -590,11 +598,19 @@ fn verify_witness_data(beacon_data: &Vec<u8>, witness_data: &Vec<u8>) -> Generic
...
+    witness_gain_dbm: i32,
+) -> i32 {
+    // beaconer and witness gain is supplied as ddbm units, convert to db
```
Comment (ke6jjj) [inline iot_verifier/src/poc.rs:608]: // beaconer and witness gain is supplied as **ddb** units, convert to db

**Review by ke6jjj — COMMENTED** [review]
Left some comments about variable names.

File: iot_verifier/src/poc.rs (hunk 1 above)
Comment (andymck) [reply iot_verifier/src/poc.rs]: fixed: 8b51e3bad7b345bf2f433bb9bae825497a795151

File: iot_verifier/src/poc.rs (hunk 2 above)
Comment (andymck) [reply iot_verifier/src/poc.rs]: removed comment, var naming gives sufficient context

**Comment by vihu** [comment]
I think this looks generally good, I'd suggest keeping the fspl function separate though (easier to test I suppose).

**Comment by andymck** [comment]
> I think this looks generally good, I'd suggest keeping the fspl function separate though (easier to test I suppose).

fixed: c96d5e600d83a330584dca92cd06bf320a9c0d8f

**Review by bbalser — APPROVED** [review]

File: iot_verifier/src/poc.rs [hunk 3]
```diff
@@ -558,10 +560,16 @@ fn verify_witness_rssi(
...
+        witness_freq,
+        distance,
+        beacon_gain,
+        witness_gain,
+    );
     // signal is submitted as DBM * 10
-    // min_rcv_signal    is plain old DBM
-    if witness_signal / 10 > min_rcv_signal as i32 {
+    // min_rcv_signal is plain old DBM
+    if (witness_signal / 10) as f64 > min_rcv_signal {
```
Comment (maplant) [inline iot_verifier/src/poc.rs:572]:
```suggestion
    if witness_signal as f64 / 10.0 > min_rcv_signal {
```

**Review by maplant — APPROVED** [review]

**Pull request merged**